### PR TITLE
Add logging options to all option types

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] Apache v2 license headers are included in each file.
 
[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/master/CHANGELOG.md

### CODE REVIEW

File: sdk/azidentity/client_certificate_credential.go [hunk 1]
```diff
@@ -96,7 +98,7 @@ func NewClientCertificateCredential(tenantID string, clientID string, certificat
 	if err != nil {
 		return nil, err
 	}
-	c, err := newAADIdentityClient(authorityHost, pipelineOptions{HTTPClient: options.HTTPClient, Retry: options.Retry, Telemetry: options.Telemetry})
```
Comment (catalinaperalta) [inline sdk/azidentity/client_certificate_credential.go:99]: The defaults should be adding after line 41 in DefaultClientCertificateCredentialOptions()

File: sdk/azidentity/authorization_code_credential.go [hunk 2]
```diff
@@ -64,7 +66,7 @@ func NewAuthorizationCodeCredential(tenantID string, clientID string, authCode s
 	if err != nil {
 		return nil, err
 	}
-	c, err := newAADIdentityClient(authorityHost, pipelineOptions{HTTPClient: options.HTTPClient, Retry: options.Retry, Telemetry: options.Telemetry})
+	c, err := newAADIdentityClient(authorityHost, pipelineOptions{HTTPClient: options.HTTPClient, Retry: options.Retry, Telemetry: options.Telemetry, Logging: options.Logging})
```
Comment (catalinaperalta) [inline sdk/azidentity/authorization_code_credential.go:69]: Same thing here. There defaults should be added after line 35 in the default options func.

File: sdk/azidentity/client_certificate_credential.go (hunk 1 above)
Comment (jhendrixMSFT) [reply sdk/azidentity/client_certificate_credential.go:99]: Fixed

File: sdk/azidentity/authorization_code_credential.go (hunk 2 above)
Comment (jhendrixMSFT) [reply sdk/azidentity/authorization_code_credential.go:70]: Fixed

**Comment by jhendrixMSFT** [comment]
I noticed that there are no config options for `DefaultAzureCredential` which seems broken.  I can add them in this PR, let me know your thoughts.

**Comment by catalinaperalta** [comment]
> I noticed that there are no config options for `DefaultAzureCredential` which seems broken. I can add them in this PR, let me know your thoughts.

That is by design, with `DefaultAzureCredential` we are literally saying you get all the defaults. I also believe this is the cross-language design for `DefaultAzureCredential`. We can bring in Scott if you want more info, but I do recall that was at least part of the reasoning, this question has come up before.

**Comment by jhendrixMSFT** [comment]
Hmm I see.  We can wait for customer feedback to see how this works in practice.

**Review by catalinaperalta — APPROVED** [review]
LGTM

**Pull request merged**